### PR TITLE
Item creation

### PR DESCRIPTION
**Pull request by tomkowalski**
This PR will: 
* Add polling on the app.js level. 
* Add item creation
* Add ability to join as a non-host (if room doesnt exist it errors a lot) 
Screenshot from 2019-04-10 22-27-39

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -106,7 +118,10 @@ class AppRouter extends Component {
                                                       viewWidth={this.state.width}
                                                       setRoomCode={this.setRoomCode}
                                                       addUserToRoom={this.addUserToRoom} /> } />
-            <Route path="/room/" render={(props) => <Room {...props} isHost={this.state.isHost} roomCode={this.state.roomCode} userId={this.state.userId} /> } /> 
+            <Route path="/room/" render={(props) => <Room {...props} isHost={this.state.isHost}
+                                                          roomCode={this.state.roomCode}
```
Comment (l-tn3314) [inline src/App.js]: probably don't need to pass this in anymore, since it'll be in `room`?

File: src/components/itemCropper.js [hunk 2]
```diff
@@ -17,7 +17,15 @@ class ItemCropper extends Component {
...
 
   moveForward() {
-    this.props.moveForward(this.state.images);
+    Promise.all(this.state.images.map((image) => {
+      fetch('http://doublewb.xyz/hci/items',
+        { method: 'POST',
+          headers: { "Content-Type" : "application/json" },
+          body: JSON.stringify({room_code: this.props.roomCode, image: image.item.str })
+        })})).then(function(){
+          this.props.setRoomCode(this.props.roomCode);
```
Comment (l-tn3314) [inline src/components/itemCropper.js]: do we need this? it's also being set at the end of the host flow?

File: src/screens/hostSetup.js [hunk 3]
```diff
@@ -119,13 +125,15 @@ class HostSetup extends Component {
...
       return <ItemCropper
+        roomCode={this.state.roomCode}
         viewWidth={this.state.width}
         viewHeight={this.state.height}
         file={this.state.blob}
         imageCallback={this.imageCallback}
         moveForward={this.moveForward}
         moveBackward={this.moveBackward}
         setTriggers={(triggers) => this.childTriggers[2] = triggers}
+        setRoomCode={this.props.setRoomCode}
```
Comment (l-tn3314) [inline src/screens/hostSetup.js]: see `itemCropper` comment

**Comment by tomkowalski** [comment]
This has been updated to work with item amounts and show them to users. Colors aren't great but are easily editable. 
Screenshot from 2019-04-11 09-50-31

File: src/screens/hostSetup.js [hunk 4]
```diff
@@ -119,13 +125,15 @@ class HostSetup extends Component {
...
     } else {
       return <ItemCropper
+        roomCode={this.state.roomCode}
         viewWidth={this.state.width}
         viewHeight={this.state.height}
         file={this.state.blob}
         imageCallback={this.imageCallback}
         moveForward={this.moveForward}
         moveBackward={this.moveBackward}
         setTriggers={(triggers) => this.childTriggers[2] = triggers}
```
Comment (l-tn3314) [inline src/screens/hostSetup.js]: `setTaxTipTotal` disappeared? it's not creating the room with the proper values, and instead setting everything to `0`

File: src/App.js [hunk 5]
```diff
@@ -22,31 +22,43 @@ class AppRouter extends Component {
...
 
   updateWindowDimensions() {
     this.setState({ width: window.innerWidth, height: window.innerHeight });
   }
 
+  updateItems() {
+    if(this.state.roomCode !== null) {
+      fetch('http://doublewb.xyz/hci/rooms/'+ this.state.roomCode)
+        .then(result => {return result.json()})
+        .then(function(result) {this.setState({ room: result })}.bind(this));
```
Comment (l-tn3314) [inline src/App.js]: i'm getting an error for `this.props.room is null` in the `Room` component when it first renders?

File: src/screens/hostSetup.js (hunk 4 above)
Comment (tomkowalski) [reply src/screens/hostSetup.js]: Fixed in the latest commit. the state wasnt updated soon enough. Fixed by sending the values with the moveCallback.

File: src/App.js (hunk 5 above)
Comment (tomkowalski) [reply src/App.js]: Fixed by letting room get queried earlier, also added more checks so it handles it better.

**Pull request merged**